yt/data_objects/particle_trajectories.py

```diff
@@ -120,7 +121,10 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
         output_field.fill(np.nan)
         for field in ("particle_position_%s" % ax for ax in "xyz"):
             for i, (fn, (time, indices, pfields)) in enumerate(sorted(my_storage.items())):
-                output_field[indices, i] = pfields[field]
+                try:
+                    output_field[indices, i] = pfields[field]
+                except ValueError:
+                    raise YTIllDefinedParticleData('The dataset contains duplicate particle index')
```

Comment:
I would change the error message to say "This dataset contains duplicate particle indices!"